cpp/tests/geometry/RGBDImage.cpp

```diff
@@ -70,396 +71,260 @@ TEST(RGBDImage, Constructor) {
-            0,   0,   0,   0,   0,   0,   237, 185, 214, 130, 32,  61,  231,
-            162, 0,   0,   0,   0,   41,  174, 233, 2,   253, 240, 190, 57,
-            93,  164, 5,   58,  161, 199, 120, 58,  20,  135, 111, 58,  222,
-            137, 38,  58,  79,  25,  59,  58,  198, 8,   20,  57,  126, 80,
-            30,  58,  5,   150, 131, 55,  249, 213, 122, 57,  101, 207, 11,
-            57,  68,  190, 82,  58,  214, 94,  32,  57};
-
-    geometry::Image depth;
-    geometry::Image color;
-
-    const int size = 5;
-
-    // test image dimensions
-    const int depth_width = size;
-    const int depth_height = size;
-    const int depth_num_of_channels = 1;
-    const int depth_bytes_per_channel = 4;
-
-    const int color_width = size;
-    const int color_height = size;
-    const int color_num_of_channels = 3;
-    const int color_bytes_per_channel = 1;
-
-    color.Prepare(color_width, color_height, color_num_of_channels,
-                  color_bytes_per_channel);
-
-    depth.Prepare(depth_width, depth_height, depth_num_of_channels,
-                  depth_bytes_per_channel);
-
-    float* const float_data = depth.PointerAs<float>();
-    Rand(float_data, depth_width * depth_height, 0.0, 1.0, 0);
-    Rand(color.data_, 130, 200, 0);
-
-    auto rgbd_image = geometry::RGBDImage::CreateFromSUNFormat(color, depth);
-
-    ExpectEQ(ref_color, rgbd_image->color_.data_);
-    ExpectEQ(ref_depth, rgbd_image->depth_.data_);
+    geometry::Image im_color;
+    EXPECT_TRUE(io::ReadImage(
+            std::string(TEST_DATA_DIR) + "/RGBD/other_formats/SUN_color.jpg",
+            im_color));
+    EXPECT_EQ(im_color.num_of_channels_, 3);
+    EXPECT_EQ(im_color.bytes_per_channel_, 1);
+
+    geometry::Image im_depth;
+    EXPECT_TRUE(io::ReadImage(
+            std::string(TEST_DATA_DIR) + "/RGBD/other_formats/SUN_depth.png",
+            im_depth));
+    EXPECT_EQ(im_depth.num_of_channels_, 1);
+    EXPECT_EQ(im_depth.bytes_per_channel_, 2);
+
+    std::shared_ptr<geometry::RGBDImage> im_rgbd =
+            geometry::RGBDImage::CreateFromSUNFormat(im_color, im_depth);
+    EXPECT_EQ(im_rgbd->color_.width_, 640);
+    EXPECT_EQ(im_rgbd->color_.height_, 480);
+    EXPECT_EQ(im_rgbd->color_.num_of_channels_, 1);
+    EXPECT_EQ(im_rgbd->color_.bytes_per_channel_, 4);
+    EXPECT_EQ(im_rgbd->depth_.width_, 640);
+    EXPECT_EQ(im_rgbd->depth_.height_, 480);
+    EXPECT_EQ(im_rgbd->depth_.num_of_channels_, 1);
+    EXPECT_EQ(im_rgbd->depth_.bytes_per_channel_, 4);
+
+    // Check data scale and truncation. These values are determined by inputs.
+    float min_val;
+    float max_val;
+    std::tie(min_val, max_val) = FloatImageMinMax(im_rgbd->color_);
+    EXPECT_FLOAT_EQ(min_val, 0.0);
+    EXPECT_FLOAT_EQ(max_val, 1.0);
+    std::tie(min_val, max_val) = FloatImageMinMax(im_rgbd->depth_);
+    EXPECT_FLOAT_EQ(min_val, 0.0);
+    EXPECT_FLOAT_EQ(max_val, 6.889);
 }
 
 TEST(RGBDImage, CreateFromNYUFormat) {
```

Comment:
We can use `GTEST_SKIP_("Reason for skipping")` in the body of the function.

Reply:
fixed